docs/developer-docs/backend/rust/intercanister.mdx

```diff
@@ -8,169 +8,72 @@ import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
-}
+```rust reference
+https://github.com/oggy-dfin/icc_rust_docs/blob/34f59ddae9fcc70173fc21927a4279757f93c51a/src/icc_rust_docs_backend/src/lib.rs#L10-L11
 ```
 
-In this code, there are three main methods: two inter-canister update methods and a query method.
-
-The first method, `async fn setup_subscribe(publisher_id: Principal, topic: String)` provides functionality for the publisher canister to subscribe to topics within the `subscriber` canister. This function is called by the publisher canister.
+## Basic ICP ledger transfer: unbounded wait calls
 
-The second method, `fn update_count(counter: Counter)` updates the counter record for each published value in a topic within the subscriber canister.
+The simplest way to interact with the ICP ledger is to use calls where the caller is willing to wait for the response for an unbounded amount of time. These calls can still fail before reaching the ledger, or even while the ledger is processing the call. but the ledger's response is guaranteed do be delivered to the caller, which is why we also refer to these calls as *guaranteed response*  calls.
 
-The third method, `fn get_count() -> u64` allows the `Counter` value to be queried and returned in a call.
+```rust reference
+https://github.com/oggy-dfin/icc_rust_docs/blob/34f59ddae9fcc70173fc21927a4279757f93c51a/src/icc_rust_docs_backend/src/lib.rs#L14-L81
+```
 
-## Deploying the canisters
+The guaranteed response delivery property ensures that the only causes of `StateUnknown` error are due to canister behavior (mismatched expectations on the return types, or the caller panicking). However, there is also a downside to unbounded wait calls: since safely upgrading a canister can only be done when the canister has no pending calls, and since unbounded wait calls provide no bound on when the call will return, the caller may be prevented from upgrading safely. This can in particular be problematic when calling untrusted canisters, such as an arbitrary ledger.
 
-Now that you've taken a look at your canisters, let's deploy them.
+We will next show how to use *bounded wait* calls instead in such cases. These calls don't guarantee that the response will be delivered, which is why we also refer to the as *best-effort response calls*, but in return don't block the caller from upgrading, and they also tolerate high system load better than unbounded wait calls. See the section on [inter-canister calls](/docs/current/developer-docs/smart-contracts/advanced-features/async-code) for more information on best-effort vs. guaranteed response calls.
 
-Open a terminal window on your local computer, if you don’t already have one open.
+## ICRC-1 transfers: bounded wait calls
 
-Then run the commands:
+We will now allow our wallet to transfer tokens on an arbitrary ICRC-1 ledger instead of just the ICP ledger. Since we in general can't trust an arbitrary ICRC-1 ledger, and we want to ensure that our canister can always be upgraded, we will use bounded wait calls.
```

Comment:
I think an untrusted ledger may not be a very good example. If I store my tokens in a ledger I have to trust it. 

One option could be that we imagine a wallet app where users can register ledgers, meaning that -- while the users registering them might be perfectly fine with trusting the ledgers -- the app itself might not be. But this might already be too complex of an example.

Reply:
We changed this example completely